xdsl/dialects/builtin.py

```diff
@@ -255,6 +255,30 @@ def print_builtin(self, printer: Printer):
         printer.print_string_literal(self.data)
 
 
+@dataclass(frozen=True)
+class SymbolNameConstr(GenericAttrConstraint[StringAttr]):
```

Comment:
This should likely be `SymbolNameConstraint`, I'll change it now. If we really want we could call it `SymbolNameAttr` instead, but it feels dishonest.